core/store/src/db/refcount.rs

```diff
@@ -137,7 +101,10 @@ fn refcount_merge<'a>(
 /// values are treated as values with reference count zero.
 pub(crate) fn get_with_rc_logic(column: DBCol, value: Option<Vec<u8>>) -> Option<Vec<u8>> {
     if column.is_rc() {
-        value.and_then(|vec| decode_value_with_rc(&vec).0.map(|v| v.to_vec()))
+        let mut value = value?;
+        decode_value_with_rc(&value).0?;
+        value.truncate(value.len() - 8);
```

Comment:
Took me some thinking to realize why this is correct even if we get an empty value due to merge: I missed the previous line with side-effectful `.0?`. Maybe expand it into explicit match? Can debug-assert that rc is possitive as well then.